Mods/EpipEncounters_7d32cb52-1cfd-4526-9b84-db4867bf9356/Story/RawFiles/Lua/Epip/QuickInventory/Filters/Equipment.lua

```diff
@@ -408,22 +420,75 @@ QuickInventory.Hooks.IsItemVisible:Subscribe(function (ev)
     ev.Visible = visible
 end)
 
--- Sort equipment by rarity.
-QuickInventory.Hooks.SortItems:Subscribe(function (ev)
-    if QuickInventory:GetSettingValue(QuickInventory.Settings.ItemCategory) == "Equipment" then
-        local rarityA, rarityB = ev.ItemA.Stats.Rarity, ev.ItemB.Stats.Rarity
-        local scoreA, scoreB
+---@param item EclItem
+---@return integer
+function ScoreLevel(item)
+    local result = item.Stats and item.Stats.Level
+    return result or 0
+end
 
-        if Item.IsArtifact(ev.ItemA) then
-            rarityA = "Artifact"
-        end
-        if Item.IsArtifact(ev.ItemB) then
-            rarityB = "Artifact"
+---@param item EclItem
+---@return integer
+function ScoreRarity(item)
+    local rarity = item.Stats and item.Stats.Rarity
+    if Item.IsArtifact(item) then
+        rarity = "Artifact"
+    end
+
+    return QuickInventory.RARITY_PRIORITY[rarity] or 0
+end
+
+---@param scoreFunction fun(EclItem):integer
+---@param itemA EclItem
+---@param itemB EclItem
+---@return integer
+function CompareItemScores(scoreFunction, itemA, itemB)
+    local scoreA = scoreFunction(itemA)
+    local scoreB = scoreFunction(itemB)
+    if scoreA == scoreB then
+        return 0
+    elseif scoreA > scoreB then
+        return 1
+    else
+        return -1
+    end
+end
+
+---@param ev Feature_QuickInventory_Hook_SortItems
+---@return integer
+function CompareEquipment(ev)
+    local sortingToScoreFunction = {
+        Level = ScoreLevel,
+        Rarity = ScoreRarity,
+    }
+
+    -- first sort by the given sorting, otherwise if tied then secondarily walk through the others
+    local initialSorting = QuickInventory:GetSettingValue(QuickInventory.Settings.Sorting)
+    local initialScoreFunction = sortingToScoreFunction[initialSorting]
+    local initialComparison = CompareItemScores(initialScoreFunction, ev.ItemA, ev.ItemB)
+    if initialComparison ~= 0 then
+        return initialComparison
+    end
+
+    for sorting, scoreFunction in pairs(sortingToScoreFunction) do
```

Comment:
Technically could lead to an undefined order if there were 3 of these functions. Personally I'd probably expose a list (and the map) of these in the feature